src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,213 @@
+    lv_draw_image_dsc_t img_dsc;
+    lv_grad_dsc_t grad;
+    lv_matrix_t matrix;
+} lv_vector_fill_dsc_t;
+
+typedef struct {
+    lv_color_t color;
+    lv_opa_t opa;
+    float width;
+    float * dash_pattern;            /* The array of consecutive pair values of the dash length and the gap length */
+    uint16_t dash_pattern_count;     /* The length of the dash_pattern array */
+    float dash_offset;               /* The shift of the starting point within the repeating dash pattern */
+    lv_vector_stroke_cap_t cap;       /* The shape to be used at the end of open subpaths */
+    lv_vector_stroke_join_t join;     /* The shape to be used at the corners of paths when they are joined */
+    uint16_t miter_limit;            /* A limit on the extent of the stroke join, the default value is 4 */
+    lv_grad_dsc_t grad;
+} lv_vector_stroke_dsc_t;
+
+typedef struct {
+    uint8_t * path;
+    uint32_t path_length;
+    uint32_t memory_size;
+} lv_vector_path_t;
+
+typedef struct {
+    float x;
+    float y;
+} lv_vector_point_t;
+
+typedef struct lv_vector_context_s lv_vector_context_t;
+
+static inline void lv_matrix_copy(lv_matrix_t * dst, lv_matrix_t * src)
+{
+    lv_memcpy(dst, src, sizeof(lv_matrix_t));
+}
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/*Path operations, implemented by different engine*/
+lv_vector_path_t * lv_vector_path_create(void);
+lv_vector_path_t * lv_vector_path_copy(lv_vector_path_t * path);
+void lv_vector_path_destroy(lv_vector_path_t * path);
+void lv_vector_path_move_to(lv_vector_path_t * path, lv_vector_point_t p);
+void lv_vector_path_line_to(lv_vector_path_t * path, lv_vector_point_t p);
+void lv_vector_path_quad_to(lv_vector_path_t * path, lv_vector_point_t p1, lv_vector_point_t p2);
+void lv_vector_path_cubic_to(lv_vector_path_t * path, lv_vector_point_t p1, lv_vector_point_t p2, lv_vector_point_t p3);
+void lv_vector_path_arc_to(lv_vector_path_t * path, lv_vector_point_t p1, lv_vector_point_t p2, float r);
+void lv_vector_path_close(lv_vector_path_t * path);
+void lv_vector_path_append_rect(lv_vector_path_t * path, lv_area_t * rect, float r);
+void lv_vector_path_append_circle(lv_vector_path_t * path, lv_vector_point_t c, float r);
+/**
+ * @param[in] start_angle The start angle of the arc given in degrees, measured counter-clockwise from the horizontal line.
+ * @param[in] sweep The central angle of the arc given in degrees, measured counter-clockwise from @p start_angle.
+ * @param[in] pie Specifies whether to draw radii from the arc's center to both of its end-point - drawn if @c true.
+ */
+void lv_vector_path_append_arc(lv_vector_path_t * path, lv_vector_point_t c, float r, uint16_t start_angle,
+                               uint16_t sweep,
+                               bool pie);
+void lv_vector_path_append_path(lv_vector_path_t * path, lv_vector_path_t * subpath);
+
+/*Context operations*/
+lv_vector_context_t * lv_vector_context_create(void);
```

Comment:
E.g. in `lv_draw_img` we call it `lv_draw_image_dsc`. So wonder if we can use the `lv_vector_dsc_t` name, and make all related functions start with `lv_vector_dsc_*`. E.g. 
```c
 lv_vector_dsc_set_fill_color(dsc_t * dsc, lv_color_t color)
```